x-pack/plugin/security/src/test/java/org/elasticsearch/integration/MultipleIndicesPermissionsTests.java

```diff
@@ -127,6 +141,61 @@ public void testSingleRole() throws Exception {
         assertHitCount(searchResponse, 1);
     }
 
+    public void testMonitorRestrictedWildcards() throws Exception {
+        IndexResponse indexResponse = index("foo", "type", jsonBuilder()
+                .startObject()
+                .field("name", "value")
+                .endObject());
+        assertEquals(DocWriteResponse.Result.CREATED, indexResponse.getResult());
+
+        indexResponse = index("foobar", "type", jsonBuilder()
+                .startObject()
+                .field("name", "value")
+                .endObject());
+        assertEquals(DocWriteResponse.Result.CREATED, indexResponse.getResult());
+
+        indexResponse = index("foobarfoo", "type", jsonBuilder()
+                .startObject()
+                .field("name", "value")
+                .endObject());
+        assertEquals(DocWriteResponse.Result.CREATED, indexResponse.getResult());
+
+        refresh();
```

Comment:
I think we need to make sure the internal security indices are created here. Maybe we create a user and a role and validate that the security indices exist

Reply:
We'll also want to look at testing the cluster health api at the indices and shards levels. Same for the cluster stats api

Reply:
Cluster Health and Stats are authorized as cluster actions `cluster:monitor/*` and will see all indices irrespective of the change proposed here.

Reply:
I went back into the time machine and figured out the API that this leniency for monitoring came from. It is the _cat/indices API. I am pretty sure it will still fail based on the way it works so I think we should:

1. add a test and validate cat indices will cause issues if the security index exists and the user is not allowed access
2. find a fix

Reply:
We dug into this over slack.
Jay pointed me to the commit that added this leniency https://github.com/elastic/x-pack/commit/0a28669fda8daa88b35ae49518521b6f38f3ccbd and we identified that it was no longer relevant as the cause got inadvertently removed by https://github.com/elastic/elasticsearch/pull/18545 .
As discussed, I have added a test for `_cat/indices/*` to validate that no unauthorized exception is thrown (when the wildcard includes the unauthorized `.security` index).